diff --git a/node_modules/foundationdb/dist/lib/apiVersion.js b/node_modules/foundationdb/dist/lib/apiVersion.js
index e9da3da..3a55eab 100644
--- a/node_modules/foundationdb/dist/lib/apiVersion.js
+++ b/node_modules/foundationdb/dist/lib/apiVersion.js
@@ -31,10 +31,16 @@ https://github.com/josephg/node-foundationdb/issues
 Until this is fixed, use FDB API version ${exports.MAX_VERSION}.
 `);
         }
-        if (headerVersion == null)
-            native_1.default.setAPIVersion(version);
-        else
-            native_1.default.setAPIVersionImpl(version, headerVersion);
+        try {
+            if (headerVersion == null)
+                native_1.default.setAPIVersion(version);
+            else
+                native_1.default.setAPIVersionImpl(version, headerVersion);
+        } catch (e) {
+            if (!e || e.message !== 'API version may be set only once') {
+                throw e;
+            }
+        }
         apiVersion = version;
     }
 }
